news/2926.bugfix

```diff
@@ -0,0 +1 @@
+Skip local directory by default for 'pip freeze' or when invoked as 'python -m pip freeze'
```

Comment:
```suggestion
Avoid listing packages from current directory, when invoked as 'python -m pip freeze'
```

Reply:
@pradyunsg, I am unsure about this comment https://github.com/pypa/pip/pull/7810#discussion_r399684766https://github.com/pypa/pip/pull/7810#discussion_r399684766

Also currently the test passes irrespective of changes due to #7864, so it just provides a pseudo confidence.